src/nmr_to_structure/prepare_input/prepare_nmr_rxn_input.py

```diff
@@ -95,64 +166,108 @@ def make_nmr_rxn_set(
             random_mol = component
             while random_mol == component:
                 random_mol = random.choice(mols_sample)
-                
-            nmr = make_nmr(mode, component, 
-                           hnmr=nmr_set.loc[random_mol]["1H_NMR_sim"]["peaks"], 
-                           cnmr=nmr_set.loc[random_mol]["13C_NMR_sim"], 
-                           hnmr_mode=hnmr_mode,
-                           token_space=token_space)
-            nmr_input_non_matching = components_tokenized + ' ' + nmr
-            src_tgt_pairs.append({'nmr_input': nmr_input_non_matching, 'smiles': NON_MATCHING_TOKEN, 'actual_smiles': random_mol})
 
-    return pd.DataFrame(src_tgt_pairs)
+            nmr = make_nmr(
+                mode,
+                random_mol,
+                hnmr=nmr_df.loc[random_mol]["1H_NMR_sim"]["peaks"],
+                cnmr=nmr_df.loc[random_mol]["13C_NMR_sim"],
+                hnmr_mode=hnmr_mode,
+                token_space=token_space,
+            )
+            nmr_input_non_matching = components_tokenized + nmr
+            src_tgt_pairs.append(
+                {
+                    "nmr_input": nmr_input_non_matching,
+                    "smiles": DEFAULT_NON_MATCHING_TOKEN,
+                }
+            )
+
+        idx += 1
+        pbar.update(len(src_tgt_pairs) - pbar.n)
+
+        if idx == len(rxns) - 1:
+            logging.warning(
+                f"Ran out of reactions to consider before reaching desired training samples. Generated {len(src_tgt_pairs)} samples instead of {n_samples} desired."
+            )
+            break
 
+    return pd.DataFrame(src_tgt_pairs)
 
-def make_nmr_mol_set(nmr_df: pd.DataFrame,
-                     mode: str, 
-                     hnmr_mode: str,
-                     token_space: str,
-                     non_matching=True, 
-                     n_mol_max: int = 7, 
-                     n_samples: int = 1000):
 
+def make_nmr_mol_set(
+    nmr_df: pd.DataFrame,
+    mode: str,
+    n_samples: int,
+    hnmr_mode: str = "range",
+    token_space: str = "shared",
+    non_matching=True,
+    n_max_mol_set_size: int = 7,
+) -> pd.DataFrame:
     mols = tuple(set(nmr_df.index))
-    src_tgt_pairs = list()
-
-    # Default mol_distribution
-    mol_distribution = [[i, n_samples] for i in range(2, n_mol_max+1)]
-
-    for n_mols, n_examples in tqdm.tqdm(mol_distribution):
+    src_tgt_pairs: List[dict] = list()
+
+    # Default mol_distribution to cover equal number of examples for the different set sizes, if non_matching is set to true -> divide by two as two examples will be added per pass
+    mol_distribution = [
+        [
+            i,
+            int(n_samples / ((n_max_mol_set_size - 1) * 2))
+            if non_matching
+            else int(n_samples / (n_max_mol_set_size - 1)),
+        ]
+        for i in range(2, n_max_mol_set_size + 1)
+    ]
```

Comment:
I don't really understand what is happening here. What does `mol_distribution` contain after that?

You can consider putting this in a separate function.

you can then probably put `int(n_samples / ((n_max_mol_set_size - 1) * 2)` in a separate variable before the for loop, with a descriptive name? It does not need to be evaluated at every iteration of the loop.